types/src/executable.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{access_path::AccessPath, state_store::state_key::StateKey};
+use aptos_crypto::HashValue;
+use std::sync::Arc;
+
+#[derive(PartialEq)]
+pub enum ExecutableDescriptor {
+    /// Possibly speculative, based on code published during the block.
+    Block(HashValue),
```

Comment:
`Published(HashValue)`?

Reply:
maybe `Publishing(HashValue)` would be even better, wdyt?

Reply:
Actually Published is better, done :-)